docs/openapi/schemas.yaml

```diff
@@ -1598,3 +1598,314 @@ ApiKeyResponse:
         apiKey:
           description: The API key
           type: string
+OpportunityStatus:
+  description: Status of an opportunity
+  type: string
+  enum:
+    - 'NEW'
+    - 'IN_PROGRESS'
+    - 'IGNORED'
+    - 'RESOLVED'
+  example: 'IN_PROGRESS'
+OpportunityType:
+  description: Type of opportunity
+  type: string
+  enum:
+    - 'Backlinks'
+    - 'Internal links'
+  example: 'Backlinks'
+OpportunityOrigin:
+  description: |
+    Origin of opportunity.
+    - AUTOMATION: generated by automation
+    - ESS_OPS: generated manually by Experience Success Engineers
+    - AI: generated by AI
+  type: string
+  enum:
+    - 'AUTOMATION'
+    - 'ESS_OPS'
+    - 'AI'
+  example: 'ESS_OPS'
+Opportunity:
+  type: object
+  properties:
+    id:
+      description: UUID of this opportunity
+      $ref: '#/Id'
+    siteId:
+      description: UUID of the site this opportunity is for
+      $ref: '#/Id'
+    auditId:
+      description: UUID of the audit result that triggered this opportunity
+      $ref: '#/Id'
+    runbook:
+      description: URL to the runbook for this opportunity
+      type: string
+      format: url
+      example: 'https://example.com/runbook/broken-backlinks'
+    type:
```

Comment:
can we've `rank` property for opportunity entity as well? that'll make it easy for UI to show only the top opportunities for each opportunity type

Reply:
^ @solaris007 
I see in the mocks there is a "sort by" in the opportunities listing. One of the options is "Highest impact", now if we have rank: 800000 (organic traffic) for one opportunity with rank: 14000 (click rate) for another opportunity, not sure which one is the highest impact and how this could be shown in the UI.

Reply:
@rpapani @iuliag as Iulia pointed out, there is no intention to introduce a rank for opportunities, due to type-specific dimensions. many opportunities will be singletons anyways and other than that i'd sort by status and updated at or number of open issues (suggestions)...

Reply:
@solaris007 @iuliag 
we'll introduce something for now specific to opportunity type. For example, for `high-organic-low-ctr/high-inorganic-high-bourncerate`, we'd like to introduce, potentialClicks, which is computed as follows

```
potentialClicks = (siteWideCTR - pageCTR) * 100 * pageviews
```
Since these opportunities are trying to increase the engagement, this potentialClicks correctly measures what can be achieved and we'll give 10 such opportunities and UI can sort accordingly.

cc: @ekremney @shrotia

Reply:
sure, we can consider adding a rank. let us first merge the current state of the data model